scanner/multiscanner.go

```diff
@@ -0,0 +1,85 @@
+package scanner
+
+import (
+	"context"
+	"fmt"
+	"sync"
+
+	"github.com/brimsec/zq/ast"
+	"github.com/brimsec/zq/filter"
+	"github.com/brimsec/zq/pkg/nano"
+	"github.com/brimsec/zq/zbuf"
+)
+
+// NewMultiScanner returns a Scanner for the logical concatenation of the
+// provided readers that filters records by filterExpr and span. Readers are
+// read sequentially, and when all records are consumed, Pull will return a nil
+// batch and nil error. If any reader returns a non-nil error, Pull will return
+// that error.
+func NewMultiScanner(ctx context.Context, readers []zbuf.Reader, filterExpr ast.BooleanExpr, span nano.Span) (Scanner, error) {
```

Comment:
We don't have to do this now but at some point it we should consider adopting the Options pattern here:
```suggestion
func NewMultiScanner(ctx context.Context, readers []zbuf.Reader, opts ...Options) (Scanner, error) {
```

Reply:
I've removed this function.